git_diff.py

```diff
@@ -0,0 +1,63 @@
+import subprocess
+import json
+from git import Repo
+
+repo = Repo("./")
+
+
+def get_staged_files():
+    try:
+        # Run the git diff command to get staged files
+        result = subprocess.run(['git', 'diff', '--name-only'],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                text=True,
+                                check=True)
+        # Return the output, which is a list of file names
+        return result.stdout.splitlines()
```

Comment:
so here returns filename only or filepath?

Reply:
The function returns the relative file path. eg. `activities/acoustic_task/acoustic_task_schema`

Reply:
I was testing the actions on a forked repo here: https://github.com/Evan8456/b2ai-redcap2rs/actions/runs/10616649473/job/29427586398

The actions workflow works well. The only time I saw issues is when we try to git tag when the tag already exists or git commit when there was nothing to new to commit. This only happened when I was rerunning the workflow on the same data dictionary tag for testing. In practice, there shouldn't be an instance where a release has no changes and an instance where a tag already exists.